src/helper/backend/PamBackend.cpp

```diff
@@ -252,8 +252,10 @@ namespace SDDM {
         QProcessEnvironment sessionEnv = m_app->session()->processEnvironment();
         const auto sessionType = sessionEnv.value(QStringLiteral("XDG_SESSION_TYPE"));
         const auto sessionClass = sessionEnv.value(QStringLiteral("XDG_SESSION_CLASS"));
-        QString tty = VirtualTerminal::path(sessionEnv.value(QStringLiteral("XDG_VTNR")).toInt());
-        m_pam->setItem(PAM_TTY, qPrintable(tty));
+        if (!sessionEnv.value(QStringLiteral("XDG_VTNR")).isEmpty()) {
```

Comment:
IMO just `sessionEnv.contains(QStringLiteral("XDG_VTNR"))`. Empty is invalid and should be caught